labgrid/util/ssh.py

```diff
@@ -0,0 +1,324 @@
+# pylint: disable=no-member
+import tempfile
+import logging
+import subprocess
+import os
+import socket
+from functools import wraps
+
+import attr
+from ..driver.exception import ExecutionError
+
+from .ports import get_free_port
+
+__all__ = ['sshmanager', 'SSHConnection', 'ForwardError']
+
+
+@attr.s
+class SSHConnectionManager:
+    """The SSHConnectionManager manages multiple SSH connections. This class
+    should not be directly instanciated, use the exported sshmanager from this
+    module instead.
+    """
+    _connections = attr.ib(
```

Comment:
Why do you use an attribute here?

Reply:
This makes debugging easier since you can see the state of private attributes as well, i.e. in backtraces. Though these are missing the `init=False` directive.